### PR TITLE
Check if links to youtube should be embedded

### PR DESCRIPTION
**Pull request by gemmaleigh**
When using the [govspeak component](http://govuk-component-guide.herokuapp.com/components/govspeak), all links to YouTube are converted to an [embedded video player](http://govuk-component-guide.herokuapp.com/components/govspeak/fixtures/with_youtube_embed/preview).

This PR allows links to YouTube which aren't converted to an embedded player.

Check for a data attribute `data-youtube-player` if its value is `off`
then don’t convert that link into an embedded player.

For example:
```
<a href="http://www.youtube.com/youtubevidid" data-youtube-player="off">
Watch this related video
</a>
```

### CODE REVIEW

**Comment by gemmaleigh** [comment]
cc. @fofr, @thehenster.

File: spec/javascripts/govuk-component/govspeak-enhance-youtube-video-links-spec.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    beforeEach(function() {
+
+      enhanceYoutubeVideoLinksFixture = '<div class="govuk-govspeak">'+
+                      '<a href="https://youtu.be/OzjogCFO4Zo">'+
+                        'This link to YouTube'+
+                      '</a>'+
+                    '</div>';
+
+      $enhanceYoutubeVideoLinksHTML = $(enhanceYoutubeVideoLinksFixture);
```
Comment (fofr) [inline spec/javascripts/govuk-component/govspeak-enhance-youtube-video-links-spec.js]: You could simplify this by going straight to `$enhanceYoutubeVideoLinksHTML` without the intermediate fixture:

eg
```
$enhanceYoutubeVideoLinksHTML = $('<div class="govuk-govspeak">'+
  '<a href="https://youtu.be/OzjogCFO4Zo">'+
    'This link to YouTube'+
  '</a>'+
'</div>');
```

File: spec/javascripts/govuk-component/govspeak-enhance-youtube-video-links-spec.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      $('body').append($enhanceYoutubeVideoLinksHTML);
+      GOVUK.enhanceYoutubeVideoLinks($(".govuk-govspeak"));
+    });
+
+    afterEach(function() {
+      $('.govuk-govspeak').remove();
```
Comment (fofr) [inline spec/javascripts/govuk-component/govspeak-enhance-youtube-video-links-spec.js]: Not necessary to change this, but you could instead call `$enhanceYoutubeVideoLinksHTML.remove()`

File: app/assets/javascripts/govuk-component/govspeak-enhance-youtube-video-links.js [hunk 3]
```diff
@@ -2,8 +2,17 @@
 //= require vendor/jquery/jquery.player.min
 
 (function ($) {
+  "use strict";
```
Comment (fofr) [inline app/assets/javascripts/govuk-component/govspeak-enhance-youtube-video-links.js]: :+1:

File: spec/javascripts/govuk-component/govspeak-enhance-youtube-video-links-spec.js (hunk 2 above)
Comment (gemmaleigh) [inline spec/javascripts/govuk-component/govspeak-enhance-youtube-video-links-spec.js]: Thanks for reviewing these 😄  I've simplified those fixtures, hopefully it makes the tests easier to read...

File: app/assets/javascripts/govuk-component/govspeak-enhance-youtube-video-links.js [hunk 4]
```diff
@@ -2,8 +2,17 @@
...
   window.GOVUK = window.GOVUK || {};
 
+  function checkPrototcolScheme(){
+    var scheme = document.location.protocol;
+    if (scheme == "file:") {
+      scheme = "https:";
+      return scheme;
+    }
```
Comment (fofr) [inline app/assets/javascripts/govuk-component/govspeak-enhance-youtube-video-links.js]: This method won't return a scheme if the scheme is not `file:`

File: spec/javascripts/govuk-component/govspeak-enhance-youtube-video-links-spec.js [hunk 5]
```diff
@@ -24,6 +24,16 @@ describe('Links to YouTube in govspeak content', function () {
       $('.media-player').remove();
     });
 
+    it("has a protocol scheme of https: or http:", function () {
```
Comment (fofr) [inline spec/javascripts/govuk-component/govspeak-enhance-youtube-video-links-spec.js]: This is a bad test – you're not testing the code within `govspeak-enhance-youtube-video-links.js`, instead the code being tested is completely within the test.

I'd expect something to mock `document.location.protocol` and test the outcome, ie the YouTube URL output, though I know this is a pain.

**Comment by fofr** [comment]
A minor comment about a test. Did you manage to test this in a browser with a frontend that uses static?

**Comment by gemmaleigh** [comment]
@fofr, yes this has been tested in the service manual frontend and works as expected.

cc. @thehenster.

**Pull request merged**